ArduCopter/ekf_check.cpp

```diff
@@ -28,21 +28,24 @@ static struct {
 // should be called at 10hz
 void Copter::ekf_check()
 {
-    // exit immediately if ekf has no origin yet - this assumes the origin can never become unset
-    Location temp_loc;
-    if (!ahrs.get_origin(temp_loc)) {
-        return;
-    }
-
-    // return immediately if motors are not armed, ekf check is disabled, not using ekf or usb is connected
-    if (!motors->armed() || ap.usb_connected || (g.fs_ekf_thresh <= 0.0f)) {
+    // return immediately if ekf check is disabled
+    if (g.fs_ekf_thresh <= 0.0f) {
         ekf_check_state.fail_count = 0;
         ekf_check_state.bad_variance = false;
         AP_Notify::flags.ekf_bad = ekf_check_state.bad_variance;
         failsafe_ekf_off_event();   // clear failsafe
         return;
     }
 
+    // Do not notify user (apart via ekf-is-unhappy Notify flag)
+    // unless motors are armed and USB is not connected and we have
+    // ever had a position
```

Comment:
Nitpicking: the comment is a bit harder to understand than it could be. I propose changing it to "*Don't trigger the EKF failsafe action if motors aren't armed, USB is connected or EKF origin hasn't been set yet.*"

Reply:
Changed.